src/Tools/AI Test Toolkit/src/TestSuite/AITTestSuiteMgt.Codeunit.al

```diff
@@ -0,0 +1,448 @@
+        ValidDatasets.Add(AITTestSuite."Input Dataset");
+
+        AITTestMethodLine.SetRange("Test Suite Code", AITTestSuite.Code);
+        if not AITTestMethodLine.FindSet() then
+            Error(NothingToRunErr);
+
+        repeat
+            CodeunitMetadata.Get(AITTestMethodLine."Codeunit ID");
+
+            // Validate test line dataset
+            if (AITTestMethodLine."Input Dataset" <> '') and (not ValidDatasets.Contains(AITTestMethodLine."Input Dataset")) then begin
+                ValidateTestLineDataset(AITTestMethodLine, AITTestMethodLine."Input Dataset");
+                ValidDatasets.Add(AITTestMethodLine."Input Dataset");
+            end;
+        until AITTestMethodLine.Next() = 0;
+    end;
+
+    local procedure ValidateSuiteDataset(var AITTestSuite: Record "AIT Test Suite")
+    begin
+        // Validate test suite
+        if AITTestSuite."Input Dataset" = '' then
+            Error(EmptyDatasetSuiteErr, AITTestSuite."Code");
+
+        if not DatasetExists(AITTestSuite."Input Dataset") then
+            Error(NoDatasetInSuiteErr, AITTestSuite."Input Dataset", AITTestSuite."Code");
+
+        if not InputDataLinesExists(AITTestSuite."Input Dataset") then
+            Error(NoInputsInSuiteErr, AITTestSuite."Input Dataset", AITTestSuite."Code");
+    end;
+
+    local procedure ValidateTestLineDataset(AITTestMethodLine: Record "AIT Test Method Line"; DatasetName: Code[100])
+    begin
+        if not DatasetExists(DatasetName) then
+            Error(NoDatasetInLineErr, DatasetName, AITTestMethodLine."Line No.");
+        if not InputDataLinesExists(DatasetName) then
+            Error(NoInputsInLineErr, DatasetName, AITTestMethodLine."Line No.");
+    end;
+
+    local procedure DatasetExists(DatasetName: Code[100]): Boolean
+    var
+        TestInputGroup: Record "Test Input Group";
+    begin
+        exit(TestInputGroup.Get(DatasetName));
+    end;
+
+    local procedure InputDataLinesExists(DatasetName: Code[100]): Boolean
+    var
+        TestInput: Record "Test Input";
+    begin
+        TestInput.Reset();
+        TestInput.SetRange("Test Input Group Code", DatasetName);
+        exit(not TestInput.IsEmpty());
+    end;
+
+    internal procedure DecreaseNoOfTestsRunningNow(var AITTestSuite: Record "AIT Test Suite")
+    begin
+        if AITTestSuite.Code = '' then
+            exit;
+        AITTestSuite.ReadIsolation(IsolationLevel::UpdLock);
+        if not AITTestSuite.Find() then
+            exit;
+        AITTestSuite.Validate("No. of Tests running", AITTestSuite."No. of Tests running" - 1);
+        AITTestSuite.Modify(true);
+        Commit();
+    end;
+
+    internal procedure ResetStatus(var AITTestSuite: Record "AIT Test Suite")
+    var
+        AITTestMethodLine: Record "AIT Test Method Line";
+        ConfirmResetStatusQst: Label 'This action will mark the run as Completed. Are you sure you want to continue ?';
+    begin
+        if Confirm(ConfirmResetStatusQst) then begin
```

Comment:
I would be here more explicit if not than exit, no identation than to code after

Reply:
done